daml-lf/kv-support/src/main/scala/com/daml/lf/kv/contracts/ContractConversions.scala

```diff
@@ -13,16 +13,17 @@ import scala.util.{Failure, Success, Try}
 object ContractConversions {
 
   def encodeContractInstance(
-      coinst: Value.VersionedContractInstance
+      coinst: Value.VersionedContractInstance,
+      agreement: String,
   ): Either[ValueCoder.EncodeError, RawContractInstance] =
     for {
-      message <- TransactionCoder.encodeContractInstance(ValueCoder.CidEncoder, coinst)
+      message <- TransactionCoder.encodeContractInstance(ValueCoder.CidEncoder, coinst, agreement)
       bytes <- SafeProto.toByteString(message).left.map(ValueCoder.EncodeError(_))
     } yield RawContractInstance(bytes)
 
   def decodeContractInstance(
       rawContractInstance: RawContractInstance
-  ): Either[ConversionError, Value.VersionedContractInstance] =
+  ): Either[ConversionError, (Value.VersionedContractInstance, String)] =
```

Comment:
Could this method return a wrapper around a string rather than a plain string ?

Reply:
a0df66ea9a2a290a011adea8c3241fab8242d151